python/ray/train/_internal/worker_group.py

```diff
@@ -174,6 +174,13 @@ def __init__(
         self.num_cpus_per_worker = num_cpus_per_worker
         self.num_gpus_per_worker = num_gpus_per_worker
         self.additional_resources_per_worker = additional_resources_per_worker
+
+        self.memory_per_worker = None
+        if self.additional_resources_per_worker is not None:
```

Comment:
Is it better to add a property in `ScalingConfig` and pass it here? Since cpu, gpu, and memory are all kwargs of `ray.remote`, intuitively we should treat them as the same.

Reply:
I'd rather leave it as an implementation detail right now. I don't expect this to be something a lot of users need, so I think keeping the `ScalingConfig` interface slim is preferable for Ray Train users.

Reply:
I think I would rather have this happen at the `ScalingConfig` level too.

It's a little weird to be pulling this out of `ScalingConfig.additional_resources_per_worker`.

[We only filter out CPU, GPU keys, so usage of `additional_resources_per_worker` is always a little awkward.](https://github.com/ray-project/ray/blob/master/python/ray/air/config.py#L237)

Reply:
Also, we'll need to update `as_placement_group_factory` anyways:

https://github.com/ray-project/ray/blob/ba3cfa0eb0daaf9ecdb17edc1c4fa4bbeb4d9dd8/python/ray/air/config.py#L251

Tune uses the converted `PlacementGroupFactory` to actually schedule the resources, and Train uses the `ScalingConfig` directly to create the `WorkerGroup`. 😢

Reply:
The reason for this is actually because [`PlacementGroup`](https://docs.ray.io/en/latest/ray-core/scheduling/placement-group.html#create-a-placement-group-reserve-resources) takes in a dictionary of resources for bundles, which hardcodes `"CPU"`, `"GPU"`, and `"memory"` keys. 

With the current PR, we don't need to modify `as_placement_group_factory` for it to work.

Reply:
I think we don't have to add a new argument in the public `ScalingConfig` api and expose it to the users. 

Just add a property to pull memory out of the resource dict should be good? like this one https://github.com/ray-project/ray/blob/ba3cfa0eb0daaf9ecdb17edc1c4fa4bbeb4d9dd8/python/ray/air/config.py#L232

Reply:
Oh right... Can we just use `_resources_per_worker_not_none` then? Rather than build the dict again from the components?